### PR TITLE
Updating Espresso XCUITest docs to ref saucectl

### PR DESCRIPTION
**Pull request by sweeneyskirt-sl**
### Description
Edited docs under /mobile-apps/automated-testing/espresso-xcuitest to remove refs to Sauce Runner and instead ref saucectl. Also added a migration map for Test Object users to compare old commands with saucectl equivalents.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md [hunk 1]
```diff
@@ -1,25 +1,172 @@
...
+| Run only tests matching the specified annotation.  | Not supported | `suites.testOptions.annotation:`  (Espresso Only) |
+| Identify a running Sauce Connect tunnel to use for secure connectivity to the cloud. | `tunnelIdentifier:` | `sauce.tunnel.id:` |
+| Specify how often (seconds) the runner should check for test results. | `checkFrequency:` | Not supported |
+| Specify the maximum length of time (minutes) the test can run. | `timeout:` | Must use CLI |
+| Specify a folder to direct the JUnit XML output. | `xmlFolder:` | Not supported |
+| Specify an alternative REST endpoint (Is this TO only? Covered by region in SL?? | `url:` | Not supported |
+| Remove shared states between tests. | Must use CLI | `suites.testOptions.clearPackageData` (Espresso Only) |
+| Set up a proxy connection. | Must use CLIL | Must use CLI |
+| Specify the concurrency to use for the test execution (up to account max). | Not supported | `sauce.concurrency:` |
+| Specify an alternative path and file to use as the configuration file. | Must use CLIL | Must use CLI |
```
Comment (FriggaHel) [inline docs/mobile-apps/automated-testing/espresso-xcuitest.md:159]:
```suggestion
| Specify an alternative path and file to use as the configuration file. | Must use CLI | Must use CLI |
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+Clone or download the [Espresso](https://github.com/saucelabs/saucectl-espresso-example) and [XCUITest](https://github.com/saucelabs/saucectl-xcuitest-example) example repos, as applicable to your project, to obtain the `saucectl` directory structure and example files for use as templates.
+
+### 3. Link your Sauce Labs account
+
+1. Generate a credentials file that `saucectl` can reference to authenticate your CLI commands.      
+    ```
+    saucectl configure
+    ```
+1. Enter your Sauce Labs `USERNAME` and `ACCESS_KEY` at the prompts.
+1. Add the resulting `credentials.yml` file in the `.sauce` folder of your project directory to your `.gitignore` file to protect it from unintended exposure.
```
Comment (FriggaHel) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md:47]: Same remark as previous one concerning that line.

File: docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+| Run only tests matching the specified size. | Not supported | `suites.testOptions.size:` (Espresso Only) |
+| Specify which package to run. | Not supported | `suites.testOptions.package:`  (Espresso Only) |
+| Run only tests matching the specified annotation.  | Not supported | `suites.testOptions.annotation:`  (Espresso Only) |
+| Identify a running Sauce Connect tunnel to use for secure connectivity to the cloud. | `tunnelIdentifier:` | `sauce.tunnel.id:` |
+| Specify how often (seconds) the runner should check for test results. | `checkFrequency:` | Not supported |
+| Specify the maximum length of time (minutes) the test can run. | `timeout:` | Must use CLI |
+| Specify a folder to direct the JUnit XML output. | `xmlFolder:` | Not supported |
+| Specify an alternative REST endpoint (Is this TO only? Covered by region in SL?? | `url:` | Not supported |
+| Remove shared states between tests. | Must use CLI | `suites.testOptions.clearPackageData` (Espresso Only) |
+| Set up a proxy connection. | Must use CLIL | Must use CLI |
```
Comment (FriggaHel) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md:150]:
```suggestion
| Set up a proxy connection. | Must use CLI | Must use CLI |
```

File: docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+| Specify the concurrency to use for the test execution (up to account max). | Not supported | `sauce.concurrency:` |
+| Specify an alternative path and file to use as the configuration file. | Must use CLIL | Must use CLI |
```
Comment (FriggaHel) [inline docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md:152]:
```suggestion
| Specify an alternative path and file to use as the configuration file. | Must use CLI | Must use CLI |
```

File: docs/mobile-apps/automated-testing/ipa-files.md [hunk 5]
```diff
@@ -60,31 +60,26 @@ You can use any of the existing methods of distribution for your iOS app, except
...
 
 ### .ipa (Optional)
 
-1. Create an empty directory name `Payload`.
-2. Move the -Runner.app to the `Payload` directory.
-3. Compress the Payload directory into an archive (.zip file) and give it a new name with .ipa file format ending.
-4. Your .ipa file is ready to be used in the `--test` parameter of our XCUITest test runner. See [Real Device Testing with Espresso and XCUITest](mobile-apps/automated-testing/espresso-xcuitest/real-devices.md) for more information.
+1. Create an empty directory name **Payload**.
+1. Move the `<AppName>.app` to the **Payload** directory.
+1. Compress the **Payload** directory into an archive (.zip file) and rename it using .ipa file format ending.
```
Comment (FriggaHel) [inline docs/mobile-apps/automated-testing/ipa-files.md:79]: Since `saucectl` `v0.47.0+` supports archiving from `.app` to `.ipa`, we may remove that, and advertise to use version `v0.47.0+`

File: docs/mobile-apps/supported-devices.md [hunk 6]
```diff
@@ -88,7 +88,7 @@ This is dedicated pool of devices just for your organization. On the mobile devi
...
+| **Requirements** | <p>Your iOS app must be:</p><p>Formatted as an .ipa file. Refer to the documentation on [how to create an .ipa file](/mobile-apps/automated-testing/ipa-files)</p><p>Uploaded and hosted in [Sauce Labs storage](/mobile-apps/app-storage.md) or installed from a remote location.</p> | <p>Your Android app must be:</p><p>Built into an .apk package/archive file.</p><p>Configured to have [internet permissions](http://developer.android.com/reference/android/Manifest.permission.html#INTERNET)</p><p>Uploaded and hosted in [Sauce Labs storage](/mobile-apps/app-storage.md) or installed from a remote location.</p>|
```
Comment (FriggaHel) [inline docs/mobile-apps/supported-devices.md:91]: > Uploaded and hosted in [Sauce Labs storage](/mobile-apps/app-storage.md) 
As `saucectl` takes care of uploading the applications, should we mention that ?

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md [hunk 7]
```diff
@@ -1,25 +1,172 @@
...
+* Sauce Labs account `USERNAME` and `ACCESS_KEY` [Look them up](https://app.saucelabs.com/user-settings)
+* Your Espresso or XCUITest apps and files
+
+## Installation & Setup
+
+`saucectl` can execute both Espresso and XCUITest tests, so the set up steps are the same regardless of which framework you are using.
+
+### 1. Install `saucectl`
+
+Begin by installing the `saucectl` CLI in the root directory of your existing mobile application automated testing project or your intended directory.
```
[the remark this answers is not in the document]
Comment (sweeneyskirt-sl) [reply docs/mobile-apps/automated-testing/espresso-xcuitest.md:24]: Okay on the root directory. With your other comment, do you mean I should not reference the cURL install example? I remember getting a comment that Android users would not use npm or brew?

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md [hunk 8]
```diff
@@ -1,25 +1,172 @@
...
+Clone or download the [Espresso](https://github.com/saucelabs/saucectl-espresso-example) and [XCUITest](https://github.com/saucelabs/saucectl-xcuitest-example) example repos, as applicable to your project, to obtain the `saucectl` directory structure and example files for use as templates.
+
+### 3. Link your Sauce Labs account
+
+1. Generate a credentials file that `saucectl` can reference to authenticate your CLI commands.      
+    ```
+    saucectl configure
+    ```
+1. Enter your Sauce Labs `USERNAME` and `ACCESS_KEY` at the prompts.
+1. Add the resulting `credentials.yml` file in the `.sauce` folder of your project directory to your `.gitignore` file to protect it from unintended exposure.
```
[the remark this answers is not in the document]
Comment (sweeneyskirt-sl) [reply docs/mobile-apps/automated-testing/espresso-xcuitest.md:45]: Hmmm. Not directing user to move the file; just to add it to .gitignore so it won't be included in the upload to sauce. Maybe I should reword if it's not clear. I will play with this one. Good feedback.

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md [hunk 9]
```diff
@@ -1,25 +1,172 @@
...
+| Choose a tablet device only. | `--tabletOnly` | Must use YAML |
+| Ensure the device is connected to a cellular network. | Not supported | Must use YAML |
+| Choose any device where the name matches the regex. | `--deviceNameQuery` | Must use YAML |
+| Specify which test class to run. | `--testsToRun <class>` | Must use YAML |
+| Specify which methods to run. | `--testsToRun <class>/<method>` | Must use YAML |
+| Exclude certain classes from the test. | Not supported  | Must use YAML |
+| Run only tests matching the specified size. | `--e size` | Must use YAML |
+| Specify which package to run. | `--e package` | Must use YAML |
+| Run only tests matching the specified annotation.  | Not supported | Must use YAML |
+| Further specify Espresso test options using supported key-value pairs. | `--e` | Not supported |
```
[the remark this answers is not in the document]
Comment (sweeneyskirt-sl) [reply docs/mobile-apps/automated-testing/espresso-xcuitest.md:106]: Thanks for the reminder...I need to add the sharding property. However, we don't have the notion of these key value pairs. We have implemented the properties we support directly (sharding, clearPackageData, etc.)

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md [hunk 10]
```diff
@@ -1,25 +1,172 @@
...
-## [Virtual Device Testing (Espresso only)](/testrunner-toolkit)
->**NOTE**: Espresso testing support on Sauce Labs virtual devices has been replaced by `saucectl`. See [Using Testrunner Toolkit](/testrunner-toolkit).
+| Configuration | Legacy YAML Properties| saucectl YAML Properties|
+| ----------------------------------- | --- | --- |
+| Specify which version of the composer API you are referencing for the configuration. | N/A | `apiversion:` |
+| Specify the framework. | Must use CLI | `kind:` |
+| Provide the location of the app to be tested. | `app:` | `espresso.app:`<br/>`xcuitest.app:` |
+| Provide the location of the test app. | `test:` | `espresso.testApp:`<br/>`xcuitest.testApp:` |
+| Identify the data center in which the real devices to test are located. | `devices.datacenter:` | `sauce.region:` |
+| Provide a name for the test. | `testname:` | `suites.name:` |
```
[the remark this answers is not in the document]
Comment (sweeneyskirt-sl) [reply docs/mobile-apps/automated-testing/espresso-xcuitest.md:136]: Ok. I will add it for the others, too.

File: docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md (hunk 2 above)
Comment (sweeneyskirt-sl) [reply docs/mobile-apps/automated-testing/espresso-xcuitest/migration.md:47]: Thanks. This file is actually going away. It is a duplicate. I just haven't pulled it out yet.

File: docs/mobile-apps/supported-devices.md (hunk 6 above)
Comment (sweeneyskirt-sl) [reply docs/mobile-apps/supported-devices.md:91]: I'll look at this. I didn't really edit this file because I was only changing the link to an outdated file. I think this page refers to apps used for both Appium and the native frameworks, so I may have to rewrite it. Thanks!

**Review by sweeneyskirt-sl — COMMENTED** [review]
Thanks for your help, Felix!

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (FriggaHel) [reply docs/mobile-apps/automated-testing/espresso-xcuitest.md:24]: The other comment was just for illustration :-)
All installation are ok since it's installed in the end !

File: docs/mobile-apps/automated-testing/espresso-xcuitest.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (FriggaHel) [reply docs/mobile-apps/automated-testing/espresso-xcuitest.md:45]: Let me rephrase to be sure it's clear.
Running `saucectl configure` will ___never___ create a `credentials.yml` file ___in the current directory___. 
User does not need to do anything except entering username/access key when it's prompted.

> 1. Add the resulting `credentials.yml` file in the `.sauce` folder of your project directory to your `.gitignore` file to protect it from unintended exposure.
=> This line can safely be totally removed.

Behind the scene, `saucectl` will actually create a `credentials.yml` file, but in a location where many programs stores their configuration (like `git` or `.vscode`), where it will not interfere with anything.
Since `.gitignore` is relative to the repository and `credentials.yml` is not within that said repository => There is no risk  of unintended exposure.

**Comment by sweeneyskirt-sl** [comment]
Thanks for all your help, @FriggaHel! I think I have addressed all of your comments, so if you wouldn't mind taking a final look! Thanks!

File: docs/mobile-apps/automated-testing/appium/appium-migration.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+#### **Appium Capabilities**
+
+Validate that the Appium capabilities you've used on TestObject will still work when you move tests to Sauce Labs; see [Appium Capabilities for Real Device Testing](/basics/test-config-annotation/test-config).
+
+### Step 4: Upload your Application to Sauce Application Storage
+
+Any mobile apps in TestObject storage will no longer be accessible after TestObject reaches end-of-life. Please upload these to Sauce Application Storage.
+
+*   For instructions on uploading your mobile apps (.apk or .ipa files) to Sauce Labs, see [Application Storage](/mobile-apps/app-storage).
+*   Update all app references in your test code capabilities. For example, in the `app` capability, you must use `storage:<app-id>` or `storage:filename=<some-app.apk>`, rather than `sauce-storage:<myapp.apk>` as you would in TestObject. See [Using Appium for Automated Mobile App Testing](/mobile-apps/appium) for more information.
```
Comment (sweeneyskirt-sl) [inline docs/mobile-apps/automated-testing/appium/appium-migration.md:74]:
```suggestion
*   Update all app references in your test code capabilities. For example, in the `app` capability, you must use `storage:<app-id>` or `storage:filename=<some-app.apk>`, rather than `sauce-storage:<myapp.apk>` as you would in TestObject. See [Using Appium for Automated Mobile App Testing](/mobile-apps/automated-testing/appium) for more information.
```

File: docs/mobile-apps/automated-testing/appium/appium-migration.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+## Modifying Real Device Test Builds
+
+Follow the steps outlined here to modify your existing test builds to utilize and reach the new unified platform.
+
+
+### Step 1: Replace Credentials
+
+ Throughout your test files, replace your TestObject credentials with your Sauce Labs credentials.
+
+  We recommend using [Environment Variables for Authentication](https://wiki.saucelabs.com/display/DOCS/Best+Practice%3A+Use+Environment+Variables+for+Authentication+Credentials) in order to reduce hard coding and protect your personal information from exposure.
```
Comment (sweeneyskirt-sl) [inline docs/mobile-apps/automated-testing/appium/appium-migration.md:41]:
```suggestion
  We recommend using [Environment Variables for Authentication](/basics/environment-variables) in order to reduce hard coding and protect your personal information from exposure.
```

File: docs/mobile-apps/automated-testing/appium/appium-migration.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+If you're using [Sauce Connect Proxy](/secure-connections/sauce-connect), our [REST API](/dev/api), or any other features in conjunction with real device testing, consult the documentation for that feature for applicable Sauce Labs endpoints.
+
+
+### Step 3: Change Device Test Options
+
+Not all device model and operating system combinations will carry over from the TestObject platform to Sauce Labs. You may need to adjust your device capabilities in your test configuration.
+
+#### **Device Allocations**
+To ensure your preferred [device allocations](https://wiki.saucelabs.com/display/DOCS/Mobile+Application+Testing+Admin+Guide) will still work when you move from TestObject to Sauce Labs, view the list of available devices for your region by logging into the Sauce Labs platform and going to **Live** > **Cross-Browser** > **Mobile Real**. For a full, non-region specific list of supported devices and platforms, see [Supported Browsers and Devices](https://saucelabs.com/platform/supported-browsers-devices).
```
Comment (sweeneyskirt-sl) [inline docs/mobile-apps/automated-testing/appium/appium-migration.md:61]:
```suggestion
To ensure your preferred [device allocations](/mobile-apps/automated-testing/appium/real-devices#configuring-appium-tests-for-real-devices) will still work when you move from TestObject to Sauce Labs, view the list of available devices for your region by logging into the Sauce Labs platform and going to **Live** > **Cross-Browser** > **Mobile Real**. For a full, non-region specific list of supported devices and platforms, see [Supported Browsers and Devices](https://saucelabs.com/platform/supported-browsers-devices).
```
